Services/UI/OJS.Services.Ui.Business/Cache/Implementations/ProblemsCache.cs

```diff
@@ -0,0 +1,39 @@
+﻿namespace OJS.Services.Ui.Business.Cache.Implementations;
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using OJS.Services.Infrastructure.Cache;
+using OJS.Services.Infrastructure.Constants;
+using OJS.Services.Infrastructure.Extensions;
+using OJS.Services.Ui.Data;
+using OJS.Services.Ui.Models.Cache;
+
+public class ProblemsCache : IProblemsCacheService
+{
+    private readonly IProblemsDataService problemsData;
+    private readonly ICacheService cache;
+
+    public ProblemsCache(
+        IProblemsDataService problemsData,
+        ICacheService cache)
+    {
+        this.problemsData = problemsData;
+        this.cache = cache;
+    }
+
+    public async Task<ICollection<ProblemCacheModel>> GetByContestId(
+        int contestId,
+        int cacheSeconds)
+        => await this.cache.Get(
+            string.Format(CacheConstants.ProblemsByContestId, contestId),
+            async () => await this.GetWithResourcesAndSubmissionTypesInProblemsByContestId(contestId),
+            cacheSeconds);
+
+    private async Task<ICollection<ProblemCacheModel>> GetWithResourcesAndSubmissionTypesInProblemsByContestId(int contestId)
+        => await this.problemsData
+            .GetAllByContest(contestId)
+            .AsNoTracking()
+            .MapCollection<ProblemCacheModel>()
+            .ToListAsync();
```

Comment:
Better make it like in `ContestsCacheService` - directly in the `GetByContestId` method, instead of a private method.
It takes more time to read the method name, than to read the logic itself (and is used only once).